dashboard/src/modules/components/OverviewComponent/index.jsx

```diff
@@ -11,48 +19,75 @@ import React, { useEffect } from "react";
 import { useDispatch, useSelector } from "react-redux";
 
 import NewRunsComponent from "./NewRunsComponent";
+import SavedRunsComponent from "./SavedRunsComponent";
 import { getDatasets } from "actions/overviewActions";
 
 const OverviewComponent = () => {
   const dispatch = useDispatch();
   const { endpoints } = useSelector((state) => state.apiEndpoint);
   const { loginDetails } = useSelector((state) => state.userAuth);
+  const [expanded, setExpanded] = React.useState(["expired", "newRuns"]);
 
   useEffect(() => {
     if (Object.keys(endpoints).length > 0) {
       dispatch(getDatasets());
     }
   }, [dispatch, endpoints, loginDetails]);
 
+  const onToggle = (id) => {
+    if (expanded.includes(id)) {
+      setExpanded(expanded.filter((item) => item !== id));
+    } else {
+      setExpanded((oldArray) => [...oldArray, id]);
+    }
```

Comment:
I agree to the perks of using `Set`. But with `useState`, initializing a Set will lead to infinite re-render of the UI

Reply:
> initializing a Set will lead to infinite re-render of the UI

Can you explain how/why this would happen?

I think the more likely problem would be missing re-renders:  if we want `setExpanded()` to cause a re-render, then I think we'll be defeated using my suggested code, because `add()` and `delete()` mutate the Set object, and so React won't notice the update.  However, this would probably be easy to fix if we wanted to (if nothing else, we can construct a new set from the modified old one).